test/test_ops.py

```diff
@@ -1000,6 +1002,43 @@ def gen_iou_check(box, expected, tolerance=1e-4):
             gen_iou_check(box_tensor, expected, tolerance=0.002 if dtype == torch.float16 else 1e-3)
 
 
+class TestMasksToBoxes:
+    def test_masks_box(self):
+        def masks_box_check(masks, expected, tolerance=1e-4):
+            out = ops.masks_to_boxes(masks)
+            torch.testing.assert_close(out, expected, rtol=0.0, check_dtype=False, atol=tolerance)
+
+        # Check for int type boxes.
+        def _get_image():
+            assets_directory = os.path.join(os.path.dirname(os.path.abspath(__file__)), "assets")
+            mask_path = os.path.join(assets_directory, "masks.tiff")
+            image = Image.open(mask_path)
+            return image
+
+        def _create_masks(image, masks):
+            for index in range(image.n_frames):
+                image.seek(index)
+                frame = np.array(image)
+                masks[index] = torch.tensor(frame)
+
+            return masks
+
+        image = _get_image()
+        expected = torch.tensor([[127, 2, 165, 40], [2, 50, 44, 92], [56, 63, 98, 100], [139, 68, 175, 104],
+                                 [160, 112, 198, 145], [49, 138, 99, 182], [108, 148, 152, 213]])
+
+        for dtype in [torch.int16, torch.int32, torch.int64]:
+            masks = torch.zeros((image.n_frames, image.height, image.width), dtype=dtype)
+            masks = _create_masks(image, masks)
+            masks_box_check(masks, expected)
+
+        image2 = _get_image()
+        for dtype in [torch.float16, torch.float32, torch.float64]:
```

Comment:
considering the expected tensor is just integer values and we use check_dtype=False, do we really need to test against all these dtypes? It seems to me that just torch.float would be enough?

I'm also not even sure the tensor we end-up with indeed has the desired dtype, it will depend on what numpy gives I think?

Reply:
I think yes, for safety as we do in other ops too E.g. have a look at TestGenBoxIou, TestBoxIou

I will defer to @datumbox

Reply:
It's different I think.

`TestGenBoxIou` doesn't make any assumption on the input/output's dtype so it's supposed to be checked against all possibilities.

Here, the output is hardcoded to be a float.

Also the expected values aren't floats

Reply:
Hmm, correct the reason is it uses `_upcast` to type cast properly. While that is used to prevent int16 and float16 overflow.
It might not happen here, So probably, we should check only for int.

Reply:
Checking only float as mentioned.

Reply:
The logic of running multiple floats is necessary for ops that do math operations that can over/under-flow. Here it's not going matter too much because all operations seem fine. This operator is so lightweight that I don't think there is a significant speed benefits from removing the multiple checks.

It's definitely not hurting us having them so if someone was to remove it on a future PR, I don't see an issue. Similarly, if one was to make a case to run always all floats through all ops, I would equally be fine.